src/x509.lisp

```diff
@@ -195,12 +195,16 @@ we are going to pass them to CFFI:WITH-POINTER-TO-VECTOR-DATA)"))
   (let ((altnames (certificate-alt-names cert)))
     (unless (cffi:null-pointer-p altnames)
       (unwind-protect
-           (flet ((alt-name-to-string (alt-name)
-                    (cffi:with-foreign-slots ((type data) alt-name (:struct general-name))
-                      (when (= type +GEN-DNS+)
-                        (alexandria:if-let ((string (try-get-asn1-string-data data '(#.+v-asn1-iastring+))))
-                          string
-                          (error "Malformed certificate: possibly NULL in dns-alt-name"))))))
+          (flet ((alt-name-to-string (alt-name)
+                   (cffi:with-foreign-slots ((type data) alt-name (:struct general-name))
+                     (case type
+                       (#. +GEN-IPADD+
+                         (let ((address (asn1-string-bytes-vector data)))
+                           (usocket:host-to-hostname address)))
+                       (#. +GEN-DNS+
```

Comment:
Why do you want read-time value here?

I think to remove it.

Reply:
`CASE` doesn't evaluate the keys, and we need the numeric value of the constants here.
(http://clhs.lisp.se/Body/m_case_.htm)